geomstats/geometry/discrete_curves.py

```diff
@@ -356,29 +376,212 @@ def normalize(self, point):
+                SRVRotationReparametrizationBundle,
+                SRVRotationReparametrizationQuotientMetric,
+            ),
+        }
+        self._sphere = Hypersphere(dim=ambient_dim - 1)
+        self._discrete_curves_with_l2 = None
+
+    def new(self, equip=True):
+        """Create manifold with same parameters."""
+        return ClosedDiscreteCurvesStartingAtOrigin(
+            ambient_dim=self.ambient_manifold.dim,
+            k_sampling_points=self.k_sampling_points,
+            equip=equip,
+        )
+
+    @property
+    def ambient_manifold(self):
+        """Manifold in which curves take values."""
+        return self.base_manifold
+
+    @property
+    def k_sampling_points(self):
+        """Number of sampling points for the discrete curves."""
+        return self.n_copies + 1
+
+    @property
+    def discrete_curves_with_l2(self):
+        """Copy of discrete curves with the L^2 metric."""
+        if self._discrete_curves_with_l2 is None:
+            self._discrete_curves_with_l2 = self.new(equip=False).equip_with_metric(
+                L2CurvesMetric
+            )
+        return self._discrete_curves_with_l2
+
+    @staticmethod
+    def default_metric():
+        """Metric to equip the space with if equip is True."""
+        return SRVMetric
+
+    def insert_origin(self, point):
+        """Insert origin as first  and last element of point."""
+        return insert_zeros(point, axis=-self.point_ndim)
+
+    def projection(self, point):
+        """Project a point from discrete curves.
+
+        Removes translation and origin, i.e. first and last elements of point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., k_sampling_points, ambient_dim]
+
+        Returns
+        -------
+        proj_point : array-like, shape=[..., k_sampling_points - 1, ambient_dim]
+        """
+        if point.shape[-2] == self.k_sampling_points - 1:
+            return gs.copy(point)
+        return (point[..., :, :] - gs.expand_dims(point[..., 0, :], axis=-2))[
+            ..., 1:, :
+        ]
+
+    def random_point(self, n_samples=1):
+        """Sample random curves.
+
+        Sampling on the sphere to avoid chaotic curves.
+        Removes translation and origin, i.e. first and last elements of point.
+
+        Parameters
+        ----------
+        n_samples : int.
+            Number of sample points.
+
+        Returns
+        -------
+        ransom_point : array-like, shape=[..., k_sampling_points - 2, ambient_dim]
+
+        """
+        sampling_times = gs.linspace(0.0, 1.0, self.k_sampling_points)
+
+        initial_point = self._sphere.random_point(n_samples)
+        initial_tangent_vec = self._sphere.random_tangent_vec(initial_point)
+
+        point = self._sphere.metric.geodesic(
+            initial_point, initial_tangent_vec=initial_tangent_vec
+        )(sampling_times)
+        return self.projection(point)
```

Comment:
Does this really generate a curve that "looks" closed ? It seems like this is the same code as the one used for open curves.